tests/test_python_emulator.py

```diff
@@ -159,3 +161,87 @@ def test_serialization():
     obj_str = dumps(batch)
     batch2 = loads(obj_str)
     assert isinstance(batch2, type(batch))
+
+
+def KS_test(
+    lhs_counts: Dict[str, int], rhs_counts: Dict[str, int], alpha: float = 0.05
+) -> None:
+    # statistical test to check of two empirical distributions are the same
+    # https://en.wikipedia.org/wiki/Kolmogorov%E2%80%93Smirnov_test#Two-sample_Kolmogorov%E2%80%93Smirnov_test
+    m = sum(lhs_counts.values())
+    n = sum(rhs_counts.values())
+    lhs_cdf = []
+    rhs_cdf = []
+    bitstrings = sorted(list(set([*lhs_counts.keys(), *rhs_counts.keys()])))
+    for bitstring in bitstrings:
+        lhs_cdf.append(lhs_counts.get(bitstring, 0) / n)
+        rhs_cdf.append(rhs_counts.get(bitstring, 0) / m)
+
+    lhs_cdf = np.cumsum(lhs_cdf)
+    rhs_cdf = np.cumsum(rhs_cdf)
+
+    c_alpha = np.sqrt(-0.5 * np.log(alpha / 2))
+
+    assert np.linalg.norm(lhs_cdf - rhs_cdf, ord=np.inf) <= c_alpha * np.sqrt(
+        (m + n) / (m * n)
+    )
+
+
+def test_bloqade_against_braket():
```

Comment:
can we get a rough estimate on how long this emulator test gonna take? so we don't keep CI running too long.

Reply:
the total time for all the tests for the emulator takes about 25 seconds or so.

Reply:
that's kinda ok, might be a bit long, is it possible to run a smaller test? (e.g 4 atoms or even 2 atoms?)

Reply:
OK I reduce time time to ~7 seconds